addons/binding/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/discovery/SomfyTahomaItemDiscoveryService.java

```diff
@@ -156,29 +234,47 @@ private boolean isSilentRollerShutter(SomfyTahomaDevice device) {
         return false;
     }
 
+    private boolean isOnOffHeatingSystem(SomfyTahomaDevice device) {
+        SomfyTahomaDeviceDefinition def = device.getDefinition();
+        for (SomfyTahomaDeviceDefinitionCommand cmd : def.getCommands()) {
+            if (cmd.getCommandName().equals(COMMAND_SET_HEATINGLEVEL)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
     private void deviceDiscovered(SomfyTahomaDevice device, ThingTypeUID thingTypeUID) {
         deviceDiscovered(device.getLabel(), device.getDeviceURL(), device.getOid(), thingTypeUID);
     }
 
     private void deviceDiscovered(String label, String deviceURL, String oid, ThingTypeUID thingTypeUID) {
-        Map<String, Object> properties = new HashMap<>(1);
+        Map<String, Object> properties = new HashMap<>(2);
         properties.put("url", deviceURL);
+        properties.put("label", label);
 
         ThingUID thingUID = new ThingUID(thingTypeUID, bridge.getThing().getUID(), oid);
 
-        if (discoveryServiceCallback.getExistingThing(thingUID) == null) {
+        Thing th = discoveryServiceCallback.getExistingThing(thingUID);
```

Comment:
You do not have to implement this logic on your own. If you create a discovery result with an existing thingUID, the existing thing will be updated with the values from your discovery automatically. So you only need to create the result here and that is it.

Reply:
ok, removed redundant check for existing thing